conductor.py

```diff
@@ -985,6 +689,99 @@ def log_time():
+    else:
+        print("File found")
+    file = request.files['file']
+    contest_name = request.form['contest_name']
+    # "if user does not select file, browser also"
+    # "submit an empty part without filename"
+    if file.filename == '':
+        return 'No selected file', 400
+    if file:  # and allowed_file(file.filename):
+        filename = os.path.join(app.config['UPLOAD_FOLDER'], secure_filename(file.filename))
+        file.save(filename)
+        with open(filename, 'r') as FO:
+            uploaded_content = FO.read()
+        try:
+            global all_contests_bristol
+            # reading it as json
+            uploaded_json = json.loads(json.dumps(ast.literal_eval(uploaded_content)))
+            # clearing all_contests_bristol
+            all_contests_bristol = []
+            for current_dic in uploaded_json:
+                # making list and dictionaries from json
+                item_dic = {'id': current_dic['id'], 'title': current_dic['title'], 'candidates': []}
+                for candidate_item in current_dic['candidates']:
+                    item_dic['candidates'].append(candidate_item)
+                all_contests_bristol.append(item_dic)
+
+            global default_audit_state
+            # updating default_audit_state key that depend on all_contests_bristol
+            default_audit_state['all_contests']['ballot_comparison'] = all_contests_bristol
+
+            global audit_types
+            # updating audit_types key that depend on all_contests_bristol
+            audit_types['ballot_comparison']['all_contests'] = all_contests_bristol
+
+            global audit_state
+            audit_state = copy.deepcopy(default_audit_state)
+            # import to help us check json file grammar
+            return jsonify(uploaded_content)
+        except:
+            return "File Format Incorrect Please use id: title: candidates:", 500
+    return "Invalid File", 500
+# JSON function end
+
+# Second Upload .json File Function
+@app.route('/upload-ballot-manifest-json2', methods=['POST'])
+def upload_second_json():
+    # "Be conservative in what you send, be liberal in what you accept"
+    # TODO: for transparency, also return the file's hash
+    if 'file' not in request.files:
+        return 'File not uploaded', 400
+    file = request.files['file']
+    contest_name = request.form['contest_name']
+    # "if user does not select file, browser also"
+    # "submit an empty part without filename"
+    if file.filename == '':
+        return 'No selected file', 400
+    if file: # and allowed_file(file.filename):
+        filename = os.path.join(app.config['UPLOAD_FOLDER'], secure_filename(file.filename))
+        file.save(filename)
+        with open(filename, 'r') as FO:
+           uploaded_content = FO.read()
+        try:
+            global reported_results_bristol
+            # reading it as json
+            uploaded_json = json.loads(json.dumps(ast.literal_eval(uploaded_content)))
```

Comment:
How does this differ from `json.loads`? If we're uploading JSON do we need `literal_eval`?